waiter/src/waiter/scheduler.clj

```diff
@@ -628,90 +628,162 @@
     :service-startup-timer (metrics/service-timer service-id "launch-overhead" "startup-time")
     :starting-instance-id->start-timestamp {}}))
 
+(defn- track-scheduling-instances
+  "Compute updated state for not-yet-scheduled instances of this service.
+   This is a private helper for update-launch-trackers, doing three things:
+   1) Record timestamps for newly-requested instances that are being scheduled.
+   2) Match (and remove) timestamps with now-scheduled (new) instances.
+   3) Drop timestamps for cancelled instance requests (due to scaling)."
+  [current-time instance-counts instance-counts' instance-scheduling-start-times
+   new-instance-ids removed-instance-ids service-id]
+  (let [;; 1) Record timestamps for newly-requested instances that are being scheduled.
+        instances-requested-delta (+ (count removed-instance-ids)
+                                     (- (:requested instance-counts')
+                                        (:requested instance-counts)))
+        all-start-times (into instance-scheduling-start-times 
+                              (repeat instances-requested-delta current-time))
+        ;; 2) Match (and remove) timestamps with now-scheduled (new) instances.
+        [matched-start-times unmatched-start-times]
+        (split-at (count new-instance-ids) all-start-times)
+        ;; 3) Drop timestamps for cancelled instance requests (due to scaling).
+        ;; NOTE: This is done *after* matching to avoid negative schedule times.
+        ;; We choose to drop the oldest timestamps, resulting in more optimistic metrics.
```

Comment:
You're right. That's a better match for Waiter's internal model of which instance it thinks it's cancelling on scale-down (we usually remove newer instances). Since we expect scale-up requests to the underlying scheduler to be handled fifo, dropping the newest seems more accurate. I'll replace `drop` with `drop-last`.